### PR TITLE
Repo | Update dev-dependencies

### PR DESCRIPTION
**Pull request by CanTuerk**
This PR will update all the dev-dependencies listed in #587 .

### CODE REVIEW

File: src/py/flwr_experimental/ops/compute/ec2_adapter.py [hunk 1]
```diff
@@ -244,9 +244,9 @@ def create_instances(
         # impaired reachability
         try:
             self._wait_until_instances_are_reachable(instance_ids=instance_ids)
-        except EC2StatusTimeout:
+        except EC2StatusTimeout as ec2statustimeout:
```
Comment (tanertopal) [inline src/py/flwr_experimental/ops/compute/ec2_adapter.py:247]:
```suggestion
        except EC2StatusTimeout as ec2_status_timeout:
```

File: src/py/flwr_experimental/ops/compute/ec2_adapter.py [hunk 2]
```diff
@@ -244,9 +244,9 @@ def create_instances(
...
             self.terminate_instances(instance_ids)
-            raise EC2CreateInstanceFailure()
+            raise EC2CreateInstanceFailure() from ec2statustimeout
```
Comment (tanertopal) [inline src/py/flwr_experimental/ops/compute/ec2_adapter.py:249]:
```suggestion
            raise EC2CreateInstanceFailure() from ec2_status_timeout
```

**Review by tanertopal — CHANGES_REQUESTED** [review]

File: src/py/flwr_experimental/ops/compute/ec2_adapter.py (hunk 1 above)
Comment (CanTuerk) [reply src/py/flwr_experimental/ops/compute/ec2_adapter.py]: Done !

File: src/py/flwr_experimental/ops/compute/ec2_adapter.py (hunk 2 above)
Comment (CanTuerk) [reply src/py/flwr_experimental/ops/compute/ec2_adapter.py]: Done !

File: pyproject.toml [hunk 3]
```diff
@@ -93,6 +93,7 @@ multi_line_output = 3
 include_trailing_comma = true
 force_grid_wrap = 0
 use_parentheses = true
+src_paths = ["src/py", "examples"]
```
Comment (danieljanes) [inline pyproject.toml:96]: Why do we need to configure this when we pass the directories `isort` should check in `test.sh` etc.?

**Review by danieljanes — CHANGES_REQUESTED** [review]
One additional comment: the import changes in `examples` are wrong, can we revert this / can we configure the tools differently?

File: dev/test.sh [hunk 4]
```diff
@@ -4,12 +4,12 @@ cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/../
...
-clang-format-10 --Werror --dry-run src/proto/flwr/proto/*                && echo "- clang-format: done" &&
-python -m isort --skip src/py/flwr/proto --check-only -rc src/py/flwr    && echo "- isort:  done" &&
-python -m black --exclude "src\/py\/flwr\/proto" --check src/py/flwr     && echo "- black:  done" &&
-python -m docformatter -c -r src/py/flwr -e src/py/flwr/proto            && echo "- docformatter:  done" &&
-python -m mypy src/py                                                    && echo "- mypy:   done" &&
-python -m pylint --ignore=src/py/flwr/proto src/py/flwr                  && echo "- pylint: done" &&
-python -m flake8 src/py/flwr                                             && echo "- flake8: done" &&
-python -m pytest -q src/py/flwr                                          && echo "- pytest: done" &&
+clang-format-10 --Werror --dry-run src/proto/flwr/proto/*                 && echo "- clang-format: done" &&
+python -m isort --check-only -s src/py/flwr/proto src/py/flwr && echo "- isort:  done" &&
```
[the remark this answers is not in the document]
Comment (CanTuerk) [reply dev/test.sh]: Done !

File: src/py/flwr/proto/transport_pb2_grpc.py [hunk 5]
```diff
@@ -1,46 +1,66 @@
 # Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
```
[the remark this answers is not in the document]
Comment (CanTuerk) [reply src/py/flwr/proto/transport_pb2_grpc.py:1]: Any generated proto files including this one were generated using the protoc.py tool with the updated grpcio-tools dependency.

File: pyproject.toml (hunk 3 above)
Comment (CanTuerk) [reply pyproject.toml:96]: We actually don't, I will remove this line from the `pyproject.toml` .

File: pyproject.toml [hunk 6]
```diff
@@ -92,6 +93,7 @@ multi_line_output = 3
 include_trailing_comma = true
 force_grid_wrap = 0
 use_parentheses = true
+known_first_party = ["flwr","flwr_example","flwr_experimental","flwr_tool"]
```
Comment (danieljanes) [inline pyproject.toml:96]: Missing spaces after the commas

File: dev/test.sh [hunk 7]
```diff
@@ -4,12 +4,12 @@ cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/../
...
-clang-format-10 --Werror --dry-run src/proto/flwr/proto/*                && echo "- clang-format: done" &&
-python -m isort --skip src/py/flwr/proto --check-only -rc src/py/flwr    && echo "- isort:  done" &&
-python -m black --exclude "src\/py\/flwr\/proto" --check src/py/flwr     && echo "- black:  done" &&
-python -m docformatter -c -r src/py/flwr -e src/py/flwr/proto            && echo "- docformatter:  done" &&
-python -m mypy src/py                                                    && echo "- mypy:   done" &&
-python -m pylint --ignore=src/py/flwr/proto src/py/flwr                  && echo "- pylint: done" &&
-python -m flake8 src/py/flwr                                             && echo "- flake8: done" &&
-python -m pytest -q src/py/flwr                                          && echo "- pytest: done" &&
+clang-format-10 --Werror --dry-run src/proto/flwr/proto/*                 && echo "- clang-format: done" &&
+python -m isort --check-only --skip src/py/flwr/proto src/py/flwr             && echo "- isort:  done" &&
```
Comment (danieljanes) [inline dev/test.sh:8]: Please align the indentation of the `echo` part with previous/following lines

**Review by danieljanes — CHANGES_REQUESTED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**